pandas/core/reshape/pivot.py

```diff
@@ -29,6 +29,11 @@ def pivot_table(data, values=None, index=None, columns=None, aggfunc='mean',
     index = _convert_by(index)
     columns = _convert_by(columns)
 
+    num_rows = data.reindex(index, axis='columns').shape[0]
```

Comment:
you are doing extra work here (e.g. the reindex), can we not do the calculation directly?

Reply:
Originally I was using `data.loc[:, index].shape[0]`, but pandas throws the "Indexing with a list with missing labels is Deprecated" warning. I use [`reindex` as it's the idiomatic approach recommended by the doc](http://pandas.pydata.org/pandas-docs/stable/whatsnew.html#indexing-with-a-list-with-missing-labels-is-deprecated).

I'm happy to make any change if you could point me to a better approach.